dio/lib/src/dio_exception.dart

```diff
@@ -211,4 +210,38 @@ class DioException implements Exception {
     }
     return msg;
   }
+
+  /// Because of [ValidateStatus] we need to consider all status codes when
+  /// creating a [DioException.badResponse].
+  static String _badResponseExceptionMessage(int statusCode) {
+    var message = '';
+    if (statusCode >= 100 && statusCode < 200) {
+      message =
+          'This is an informational response - the request was received, continuing processing';
+    } else if (statusCode >= 200 && statusCode < 300) {
+      message =
+          'The request was successfully received, understood, and accepted';
+    } else if (statusCode >= 300 && statusCode < 400) {
+      message =
+          'Redirection: further action needs to be taken in order to complete the request';
+    } else if (statusCode >= 400 && statusCode < 500) {
+      message =
+          'Client error - the request contains bad syntax or cannot be fulfilled';
+    } else if (statusCode >= 500 && statusCode < 600) {
+      message =
+          'Server error - the server failed to fulfil an apparently valid request';
+    } else {
+      message =
+          'A response with a status code that is not within the range of inclusive 100 to exclusive 600'
+          'is a non-standard response, possibly due to the server\'s software.';
+    }
+
+    return ''
```

Comment:
Maybe `StringBuffer` would be a good choice at here.

Reply:
I'm not really aware of the benefits of a `StringBuffer`. Can you explain why it would improve this code?

Reply:
> I'm not really aware of the benefits of a `StringBuffer`. Can you explain why it would improve this code?

You just don't have to constantly write `\n` then.